### ISSUE TITLE
Generate example `epi_archive` on demand.

### ISSUE THREAD

**Issue by dajmcdon** [cmu-delphi/epiprocess#125]
`archive_cases_dv_subset` needs to be regenerated or reworked to form an `epi_archive` from more basic parts on demand; it currently appears to load with the old class definition, missing the extra "\n", plus some bad srcrefs that may make user debugging of tweaked examples harder:
```
archive_cases_dv_subset$print
<srcref: file "C:/Users/chloe/Desktop/Summer2022/epiprocess/R/archive.R" chars 162:19 to 187:11>
<environment: 0x55c5b02d6358>
Warning message:
In getSrcLines(srcfile, x[7L], x[8L]) :
  restarting interrupted promise evaluation
> body(archive_cases_dv_subset$print)
{
    cat("An `epi_archive` object, with metadata:\n")
    cat(sprintf("* %-9s = %s\n", "geo_type", self$geo_type))
    cat(sprintf("* %-9s = %s\n", "time_type", self$time_type))
    if (!is.null(self$additional_metadata)) {
        sapply(self$additional_metadata, function(m) {
            cat(sprintf("* %-9s = %s\n", names(m), m))
        })
    }
    cat("----------\n")
    cat(sprintf("* %-14s = %s\n", "min time value", min(self$DT$time_value)))
    cat(sprintf("* %-14s = %s\n", "max time value", max(self$DT$time_value)))
    cat(sprintf("* %-14s = %s\n", "min version", min(self$DT$version)))
    cat(sprintf("* %-14s = %s\n", "max version", max(self$DT$version)))
    cat("----------\n")
    cat(sprintf("Data archive (stored in DT field): %i x %i\n", 
        nrow(self$DT), ncol(self$DT)))
    cat("----------\n")
    cat(sprintf("Public methods: %s", paste(names(epi_archive$public_methods), 
        collapse = ", ")))
}
``` 
The quick fix is to just regenerate the object + file an Issue to later figure out how to generate the example epi_archive on demand or otherwise ensure that it is current with the implementation of epi_archive.

**Comment by dajmcdon** [comment]
Originally described in #85

**Comment by brookslogan** [comment]
The "fix" above seemed to work and avoid the need for `@include` tags when I tested it off of https://github.com/cmu-delphi/epiprocess/commit/8f36d44c2379a847da27a642e335eee7a4b17c48, as it seemed to allow me to load or document the package without errors.  However, now, when I already have `epiprocess` loaded via devtools and I modify the `epi_archive` class to try to assign to new fields I've forgotten to initially set to `NULL`, I get complaints that I "cannot add bindings to a locked environment", even after fixing the problem.  I have to `unloadNamespace("epiprocess")` first to have it successfully reload.  Somehow, it seems to successfully use the new `epi_archive` if loading for the first time, but the old `epi_archive` if reloading.

Maybe my statement about not needing `@include`s is wrong.  Maybe my method of applying `delayedAssign` is wrong.  Maybe there's a `devtools` bug.

**Comment by brookslogan** [comment]
To summarize the actual issue here:
1. Start from a non-buggy package.
2. Introduce a bug that allows the package to load, but causes the promise to produce an error when evaluated.
3. `load_all()` the package.  It will successfully load.
4. `load_all()` the package again.  It will emit an error message about the bug.
5. `load_all()` the package again.  It will emit an error message about the bug + a warning about `restarting interrupted promise evaluation`.
6. To escape the cycle without restarting, `unloadNamespace("pkgname")` seems to work, although I'm not sure whether it's the "proper" function to call here.

Note that `document()` appears to `load_all()` twice, so it do both steps 3&4 together.

This could be pretty surprising while developing.  Maybe we can add a `.onLoad` function that tests creation of a 1-row archive to help prevent it, or make the archive promise into a regular object.

------

A separate issue with the current implementation is that the memory use is going to be about double what it needs to be, as both `archive_cases_dv_dt` and `archive_cases_dv` will be evaluated and stick around.  There's probably a way around this, but it raises yet another issue which is that when memory _actually_ matters (it doesn't for these small examples), the user may want the ability to "move" `x` into the archive / share `x` with the archive, not cloning.

**Comment by brookslogan** [comment]
Resolved with #101.
